### ISSUE TITLE
ESLint parsing error for instanceof syntax

### ISSUE THREAD

**Issue by maxammann** [gajus/eslint-plugin-flowtype#342]
Heyho,

If you check for instanceof with a generic type you get an eslint parsing error.

`if (Array.isArray(extrasPayload.data) && extrasPayload.data instanceof Array<ExtraModel>) {`

Results in:
```
  30:93  error  Parsing error: Unexpected token

  28 |     const extrasPayload = await extrasEndpoint.loadData(dispatch, state.extras, {city, language})
  29 |
> 30 |     if (Array.isArray(extrasPayload.data) && extrasPayload.data instanceof Array<ExtraModel>) {
```

I'm using the babel-eslint parser.

**Comment by pnevyk** [comment]
Hi @maxammann 

Parser errors are related to the parser, we work with information which parser gives us, but if it cannot parse the source code, it can't give us anything.

In your case however, it is not valid javascript code. Babel-eslint strips flow type annotations, like

```javascript
let x: number = 3;
//  -> 
let x = 3;
```

But in `extrasPayload.data instanceof Array<ExtraModel>` flow syntax is part of the statement and hence the statement is not in valid javascript syntax. If you need to ensure that `extrasPayload.data` is of type Array\<ExtraModel>, you have to write a js function which actually test the values of the array and has appropriate flow annotations, so e.g.
```javascript
function isArrayOfExtraModels(data: Array<mixed>): boolean {
    // check if data is of type Array<ExtraModel>
}
```

And then:

```javascript
if (Array.isArray(extrasPayload.data) && isArrayOfExtraModels(extrasPayload.data)) {
     // ...
}
```

**Comment by maxammann** [comment]
Makes sense aaand it was my fault..

My IDE did not show the flow message and only the eslint error. If I run `yarn flow` it shows an error. So yes it's not supported by flow and it makes sense that it isn't imo :)

Thanks and sorry for the invalid report.
